### ISSUE TITLE
Update: Passwords ending in numbers are the issue, not special characters.

### ISSUE THREAD

**Issue by darkdragon2242** [MrSonicMaster/vHackXT-MoneyBot#12]
I have some $$ in my password on another account of mine and I can't login with those in the password, is there a workaround for that?

I did get the bot working on the account I was having issues with before though

**Comment by MrSonicMaster** [comment]
Hmm special characters shouldn't cause an error, did it just say invalid credentials, or did it actually give an error? If it did, please send it here so I can try to fix it.

**Comment by darkdragon2242** [comment]
It says invalid credentials, but I can login from the game just fine.

**Comment by MrSonicMaster** [comment]
Hmm I'll look into this, thanks for pointing it out.

**Comment by MrSonicMaster** [comment]
I have an idea, I possibly would need to escape the text so it is supported in the request, but I would need to test this idea first.

**Comment by MrSonicMaster** [comment]
I created a new account with the username as all supported special characters (which are [] and {}) and a password with every single special character I could fit in the 16 characters max, and the bot successfully logged me in perfectly with no errors using the same exact one that is  uploaded to Github. Maybe you mistyped one character by accident? I don't think its a bug with the bot.

**Comment by darkdragon2242** [comment]
![0cb3e2910ebb48cd967bfc149082e11a](https://cloud.githubusercontent.com/assets/26208184/24077484/ad736054-0c1c-11e7-9004-273ab9cfcbe5.png)
I will see if I can duplicate the issue, but no I haven't typed the password wrong.

**Comment by darkdragon2242** [comment]
Ok so maybe its just my format for username and password, My username has a cap for the first letter but no special characters, the password just has some $$ in it. I was able to create another account with a password also containing 2 $$ and a few numbers and duplicated the issue.
{
  "userName": "Branismo",
  "userPass": "D$$G3lg5t1",
  "uHashStr": "This_Doesn't_Seem_To_Matter_So_Don't_Edit_This"
}

Edit: you can copy and paste this login into nox and it will work just fine.

**Comment by MrSonicMaster** [comment]
You can attempt to login with this example account that I created, and it works perfectly fine
{ 
	"userName": "[]{}[]{}[]",
	"userPass": "!@#$%^&*",
	"uHashStr": "This_Doesn't_Seem_To_Matter_So_Don't_Edit_This"
}

It has as many special characters as I could fit, and it works perfectly fine.

**Comment by darkdragon2242** [comment]
Yes the one you shared just now works fine, but did you try the one I shared?

For some reason it doesn't work, and I'm not sure why, I am trying to find the root of the issue now.

Edit: I'm not sure if I found the issue or not but I think it may have something to do with the $ in the second spot of the password or because it follows a letter? IDK, but I can't test now because it's not letting me create more accounts.

I tried with the password A$A1nf1f1, I guess it could possibly be numbers as well, I hadn't thought about that until now, but I see all the passwords I have used end in a number.

**Comment by darkdragon2242** [comment]
For some reason it doesn't work, and I'm not sure why, I am trying to find the root of the issue now.

**Comment by darkdragon2242** [comment]
Ok so after testing what works and what doesn't I have made 2 accounts, one works with the bot, the other does not.

**Working bot account**
```
{
"userName": "Branismon",
"userPass": "D$$G3lg5t",
"uHashStr": "This_Doesn't_Seem_To_Matter_So_Don't_Edit_This"
}
```
![fac5a7b1716844b4bccd3a8b42336e56](https://cloud.githubusercontent.com/assets/26208184/24078419/1f45db08-0c3b-11e7-8e15-25b11fe10cd7.png)

**Not working bot account**
```
{
"userName": "Branismo",
"userPass": "D$$G3lg5t1",
"uHashStr": "This_Doesn't_Seem_To_Matter_So_Don't_Edit_This"
}
```
![deb0d369d809462291a087e5e81f60af](https://cloud.githubusercontent.com/assets/26208184/24078427/73c7a7c4-0c3b-11e7-8a67-49729d7e7d9a.png)

You can successfully log in to both accounts from the app itself, however the one with the 1 at the end of the password does not work. At first I assumed it was due to special characters, however now I see that I was wrong. What is really trippy is I also made an account with the password "_password1_" and it worked fine, so it looks like more testing to get the root issue is needed.

**Comment by MrSonicMaster** [comment]
Hmm thanks for pointing this out, I can't begin to imagine why a password ending in a number might cause it to fail to login from my code. I'll look into it.

**Comment by darkdragon2242** [comment]
So have you managed to find any solutions or workarounds for this?

**Comment by MrSonicMaster** [comment]
I can't see any reason that it might cause an issue, I have looked through the android code and my code and it doesn't seem to have any problems

**Comment by darkdragon2242** [comment]
Is it possible to change the way the variables are stored so that the password string is read differently? I can't find any issues either but for some reason it just will not work.

**Comment by MrSonicMaster** [comment]
I'm sorry for my away time, I'll get back to working on this project and responding. And I'll look into the issue more in a bit
